huaweicloud/services/live/resource_huaweicloud_live_domain.go

```diff
@@ -94,132 +94,180 @@ func ResourceDomain() *schema.Resource {
+		"domain_type":           d.Get("type"),
+		"region":                region,
+		"service_area":          utils.ValueIgnoreEmpty(d.Get("service_area")),
+		"enterprise_project_id": utils.ValueIgnoreEmpty(d.Get("enterprise_project_id")),
+	}
+
+	return params
+}
+
 func resourceDomainRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	c := meta.(*config.Config)
-	region := c.GetRegion(d)
-	client, err := c.HcLiveV1Client(region)
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
 	if err != nil {
-		return diag.Errorf("error creating Live v1 client: %s", err)
+		return diag.Errorf("error creating Live client: %s", err)
 	}
 
-	domain := d.Id()
-	response, err := client.ShowDomain(&model.ShowDomainRequest{
-		Domain:              &domain,
-		EnterpriseProjectId: utils.StringIgnoreEmpty(c.GetEnterpriseProjectID(d)),
-	})
+	respBody, err := GetDomain(client, d.Id())
 	if err != nil {
 		return common.CheckDeletedDiag(d, err, "error retrieving Live domain")
 	}
 
-	if response.DomainInfo == nil || len(*response.DomainInfo) != 1 {
-		return diag.Errorf("error retrieving Live domain")
-	}
-	r := *response.DomainInfo
-	detail := r[0]
-
 	mErr := multierror.Append(
 		d.Set("region", region),
-		d.Set("name", detail.Domain),
-		d.Set("type", utils.MarshalValue(detail.DomainType)),
-		d.Set("status", utils.MarshalValue(detail.Status)),
-		d.Set("ingest_domain_name", detail.RelatedDomain),
-		d.Set("cname", detail.DomainCname),
-		d.Set("service_area", flattenServiceAreaAttribute(detail.ServiceArea)),
-		d.Set("enterprise_project_id", detail.EnterpriseProjectId),
-		d.Set("is_ipv6", detail.IsIpv6),
+		d.Set("name", utils.PathSearch("domain", respBody, nil)),
+		d.Set("type", utils.PathSearch("domain_type", respBody, nil)),
+		d.Set("status", utils.PathSearch("status", respBody, nil)),
+		d.Set("ingest_domain_name", utils.PathSearch("related_domain", respBody, nil)),
+		d.Set("cname", utils.PathSearch("domain_cname", respBody, nil)),
+		d.Set("service_area", utils.PathSearch("service_area", respBody, nil)),
+		d.Set("enterprise_project_id", utils.PathSearch("enterprise_project_id", respBody, nil)),
+		d.Set("is_ipv6", utils.PathSearch("is_ipv6", respBody, nil)),
 	)
 
 	return diag.FromErr(mErr.ErrorOrNil())
 }
 
-func flattenServiceAreaAttribute(serviceArea *model.DecoupledLiveDomainInfoServiceArea) string {
-	if serviceArea == nil {
-		return ""
+func GetDomain(client *golangsdk.ServiceClient, domainName string) (interface{}, error) {
+	httpUrl := "v1/{project_id}/domain"
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = fmt.Sprintf("%s?domain=%s", getPath, domainName)
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
```

Comment:
```suggestion
	if err != nil {
		return nil, golangsdk.ErrDefault404{}
	}
```
If there are no other error codes, remove this block